stl/inc/compare

```diff
@@ -340,9 +341,503 @@ struct compare_three_way {
+    };
+
+    template <class _Ty1, class _Ty2>
+    concept _Can_compare_three_way = requires(_Ty1&& _Left, _Ty2&& _Right) {
+        static_cast<strong_ordering>(compare_three_way{}(_Left, _Right));
+    };
+
+    class _Cpo {
+    private:
+        enum class _St { _None, _Adl, _Floating, _Three };
+
+        template <class _Ty1, class _Ty2>
+        _NODISCARD static _CONSTEVAL _Choice_t<_St> _Choose() noexcept {
+            if constexpr (!same_as<decay_t<_Ty1>, decay_t<_Ty2>>) {
+                return {_St::_None};
+            } else if constexpr (_Has_ADL<_Ty1, _Ty2>) {
+                return {_St::_Adl,
+                    noexcept(static_cast<strong_ordering>(strong_order(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
+            } else if constexpr (floating_point<decay_t<_Ty1>>) {
+                return {_St::_Floating, true};
+            } else if constexpr (_Can_compare_three_way<_Ty1, _Ty2>) {
+                return {_St::_Three, noexcept(static_cast<strong_ordering>(
+                                         compare_three_way{}(_STD declval<_Ty1&>(), _STD declval<_Ty2&>())))};
+            } else {
+                return {_St::_None};
+            }
+        }
+
+        template <class _Ty1, class _Ty2>
+        static constexpr _Choice_t<_St> _Choice = _Choose<_Ty1, _Ty2>();
+
+    public:
+        // clang-format off
+        template <class _Ty1, class _Ty2>
+            requires (_Choice<_Ty1, _Ty2>._Strategy != _St::_None)
+        _NODISCARD constexpr strong_ordering operator()(_Ty1&& _Left, _Ty2&& _Right) const
+            noexcept(_Choice<_Ty1, _Ty2>._No_throw) {
+            // clang-format on
+            constexpr _St _Strat = _Choice<_Ty1, _Ty2>._Strategy;
+            if constexpr (_Strat == _St::_Adl) {
+                return static_cast<strong_ordering>(strong_order(_Left, _Right));
+            } else if constexpr (_Strat == _St::_Floating) {
+                using _Floating_type = decay_t<_Ty1>;
+                using _Traits        = _Floating_type_traits<_Floating_type>;
+                using _Uint_type     = typename _Traits::_Uint_type;
+
+                auto _Left_uint  = _STD bit_cast<_Uint_type>(_Left);
+                auto _Right_uint = _STD bit_cast<_Uint_type>(_Right);
+
+                // 1. Ultra-fast path: equal representations are equal.
+                if (_Left_uint == _Right_uint) {
+                    return strong_ordering::equal;
+                }
+
+                // 2. Examine the sign bits.
+                const bool _Left_negative  = (_Left_uint & _Traits::_Shifted_sign_mask) != 0;
+                const bool _Right_negative = (_Right_uint & _Traits::_Shifted_sign_mask) != 0;
+
+                // 3. Fast path: any negative value is less than any positive value.
+                if (_Left_negative != _Right_negative) {
+                    return _Left_negative ? strong_ordering::less : strong_ordering::greater;

```

Comment:
I believe that's far too clever - it has to reverse right/left because "negative" being "true" being "1" is the opposite of what we want. As written, the single line of code matches the comment. No change made here.